release notes/v0.40.0.md

```diff
@@ -0,0 +1,272 @@
+
+Additionally as previously mentioned there were a bunch of changes to goja including adding class support which also needed to be integrated.
+
+The k6 team is hopeful that this will land in the next version v0.41.0!
+
+A future breaking change (⚠) is that it will no longer be possible to be using CommonJS style exports along import/export syntax in the *same* file.
+
+```javascript
+import http from "k6/http"
+
+exports.default = function() {} // this will start to error out
+```
+
+It will still be possible to import a CommonJS module and `require` a ES module, or use `require` in an ES module. Having a dependency cycle mixing and matching CommonJS and ESM is also unlikely to work properly, but *might* do so in particular cases.
+
+This is really expected to have never been done by anyone as there isn't really much of a reason for this, but is currently supported due to babel transpiling everything to CommonJS behind the scenes.
+
+## Refactoring metrics
+
+The refactoring of metrics has been under way and there are both a [PR](https://github.com/grafana/k6/pull/2594) for a more performant internal representation. This unfortunately took longer to get to a stable state and given the size and the type of change the k6 team decided to *not* release it very near the end of the cycle. It also doesn't have any noticeable change for most users. Instead of that it will be merged early on in the v0.41.0 release cycle and then more changes to make use of it will be made through the release cycle.
+
+Some of hose changes include adding support for non-indexable tags which also has a read [PR](https://github.com/grafana/k6/pull/2654). This change also is really important for historical reasons connected with how `name` the tag works. As such it also was going to be needed to be merged in order to release the internal metric refactor.
+
+⚠ Future breaking change: As part of the many changes and experiments we found out that we can keep `url` as indexable tag. Previously the plan is that it *along* `vu` and `iter` will become non-indexable. But the `url` tag is *heavily* used and enabled by default. Because of that (and other internal reasons) the new change will be that `url` will stay indexable, but if used with `name` will be overwritten to have the same value as `name`. Arguably this is what most users would want in the case when they are using `name`. For the ones that do not we plan on adding a *non* indexable `raw_url` tag. As such we no longer will be [printing a warning when `url` is used](https://github.com/grafana/k6/pull/2655) in thresholds.
```

Comment:
```suggestion
⚠ Future breaking change: As part of the many changes and experiments, we found out that we can keep `url` as an indexable tag. Previously the plan was that it *along* `vu` and `iter` will become non-indexable. However, the `url` tag is *heavily* used and enabled by default. Because of that (and other internal reasons), the new change will be that `url` will stay indexable, but if used with `name` will be overwritten to have the same value as `name`. Arguably this is what most users would want in the case when they are using `name`. We plan to add a *non* indexable `raw_url` tag for those that do not. As such, we no longer will be [printing a warning when `url` is used](https://github.com/grafana/k6/pull/2655) in thresholds.
```